### PR TITLE
Improve reliability of promo registration stats sync. Closes #2478

### LINKED ISSUE #2478: Refactor Sync::ChannelPromoRegistrationsStatsJob
Currently it's a long running job. This refactor will include retry and increase concurrency by breaking up

```
  def perform
    active_publisher_ids = Publisher.not_suspended
    promo_registrations = PromoRegistration.channels_only.where(publisher_id: active_publisher_ids)
    Promo::RegistrationsStatsFetcher.new(promo_registrations: promo_registrations, update_only: true).perform
  end
```

Into something like

```
  def perform
    active_publisher_ids = Publisher.not_suspended
    promo_registration_ids = PromoRegistration.channels_only.where(publisher_id: active_publisher_ids).pluck(:id)
    ids = []
    promo_registration_ids.each do |promo_registration_id|
      ids << promo_registration_id
      if ids.count >= 50
        Sync::PromoRegistrationStats.new(promo_registration_ids: ids).perform
        ids = []
      end
    end
    Sync::PromoRegistrationStats.new(promo_registration_ids: ids).perform if ids.count > 0
  end
```

### PR DESCRIPTION
**Pull request by yachtcaptain23**

### CODE REVIEW

File: app/jobs/sync/channel_promo_registrations_stats_job.rb [hunk 1]
```diff
@@ -4,7 +4,15 @@ class Sync::ChannelPromoRegistrationsStatsJob < ApplicationJob
...
+    promo_registration_ids = PromoRegistration.channels_only.where(publisher_id: active_publisher_ids).pluck(:id)
+    ids = []
+    promo_registration_ids.each do |promo_registration_id|
+      ids << promo_registration_id
+      if ids.count >= 50
+        Sync::PromoRegistrationStats.perform_async(ids)
+        ids = []
+      end
+    end
+    Sync::PromoRegistrationStats.perform_async(ids).perform if ids.count >= 50
```
Comment (corymcdonald) [inline app/jobs/sync/channel_promo_registrations_stats_job.rb]:
```suggestion
    Sync::PromoRegistrationStats.perform_async(ids).perform if ids.count >= 0
```

File: app/jobs/sync/channel_promo_registrations_stats_job.rb [hunk 2]
```diff
@@ -4,7 +4,15 @@ class Sync::ChannelPromoRegistrationsStatsJob < ApplicationJob
...
   def perform
     active_publisher_ids = Publisher.not_suspended
-    promo_registrations = PromoRegistration.channels_only.where(publisher_id: active_publisher_ids)
-    Promo::RegistrationsStatsFetcher.new(promo_registrations: promo_registrations, update_only: true).perform
+    promo_registration_ids = PromoRegistration.channels_only.where(publisher_id: active_publisher_ids).pluck(:id)
+    ids = []
+    promo_registration_ids.each do |promo_registration_id|
+      ids << promo_registration_id
+      if ids.count >= 50
+        Sync::PromoRegistrationStats.perform_async(ids)
```
Comment (corymcdonald) [inline app/jobs/sync/channel_promo_registrations_stats_job.rb]: Syntax error here - Should be `PromoRegistrationStatsJob`

Comment (yachtcaptain23) [inline app/jobs/sync/channel_promo_registrations_stats_job.rb]:
```suggestion
        Sync::PromoRegistrationStatsJob.perform_async(ids)
```

File: app/jobs/sync/channel_promo_registrations_stats_job.rb [hunk 3]
```diff
@@ -4,7 +4,15 @@ class Sync::ChannelPromoRegistrationsStatsJob < ApplicationJob
...
+    promo_registration_ids = PromoRegistration.channels_only.where(publisher_id: active_publisher_ids).pluck(:id)
+    ids = []
+    promo_registration_ids.each do |promo_registration_id|
+      ids << promo_registration_id
+      if ids.count >= 50
+        Sync::PromoRegistrationStatsJob.perform_async(ids)
+        ids = []
+      end
+    end
+    Sync::PromoRegistrationStats.perform_async(ids).perform if ids.count > 0
```
Comment (yachtcaptain23) [inline app/jobs/sync/channel_promo_registrations_stats_job.rb]:
```suggestion
    Sync::PromoRegistrationStatsJob.perform_async(ids).perform if ids.count > 0
```

File: app/jobs/sync/channel_promo_registrations_stats_job.rb [hunk 4]
```diff
@@ -4,7 +4,15 @@ class Sync::ChannelPromoRegistrationsStatsJob < ApplicationJob
...
+    promo_registration_ids = PromoRegistration.channels_only.where(publisher_id: active_publisher_ids).pluck(:id)
+    ids = []
+    promo_registration_ids.each do |promo_registration_id|
+      ids << promo_registration_id
+      if ids.count >= 50
+        Sync::PromoRegistrationStatsJob.perform_async(ids)
+        ids = []
+      end
+    end
+    Sync::PromoRegistrationStatsJob.perform_async(ids).perform if ids.count > 0
```
Comment (yachtcaptain23) [inline app/jobs/sync/channel_promo_registrations_stats_job.rb]:
```suggestion
    Sync::PromoRegistrationStatsJob.perform_async(ids) if ids.count > 0
```

**Pull request merged**